### DISCUSSION TITLE
Installing browser extensions

### DISCUSSION THREAD

**Discussion by ajtruckle** [MicrosoftEdge/WebView2Feedback#4145] [Q&A] [answered]
Win32

What is basic approach for managing extensions? How are they installed etc? Is there sample code?

**Comment by victorhuangwq** [comment]
Hi! We recently promoted browser extensions API in the pre-release SDK. You can try it out in the pre-release SDK 
https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes?tabs=dotnetcsharp#promotions-1

**Reply by Monarkos** [reply]
How do I use this in MAUI C#?

**Reply by ajtruckle** [reply]
@victorhuangwq it is still not clear how to use it. Eg. Where does one get the browser extension itself to install it?

**Reply by victorhuangwq** [reply, marked as answer]
Hey @ajtruckle you would have to obtain your own unpacked extension, and then call the https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2profile7?view=webview2-1.0.2194-prerelease&preserve-view=true#addbrowserextension
on it. 

Here's a sample code: https://github.com/MicrosoftEdge/WebView2Samples/blob/main/SampleApps/WebView2APISample/ScenarioExtensionsManagement.cpp

There's also a thread here as well: https://github.com/MicrosoftEdge/WebView2Feedback/issues/3694

**Reply by Monarkos** [reply]
Both ways it's not clear how to do it!
And the method for webview2 it's only on WinRT, so not possible to use on MAUI, I Would prefer intall from unload package when webview2 is created

**Reply by ajtruckle** [reply]
@Monarkos I assumed we could manually download an extension via Edge, or some how get it’s official uri path (eg. for Print Friendly PDF) and install it. 

I did not realise I had to create an extension myself, which I have no interest in doing. 

As for MAUI, I would look here and ask in the comments of the videos. The maintainer there is very approachable and does excellent video tutorials for MAUI:

https://youtube.com/@jfversluis?feature=shared

**Reply by victorhuangwq** [reply]
@ajtruckle as mentioned in the https://github.com/MicrosoftEdge/WebView2Feedback/issues/3694, there are workarounds:

https://github.com/MicrosoftEdge/WebView2Feedback/issues/3694#issuecomment-1993649183
https://github.com/MicrosoftEdge/WebView2Feedback/issues/3694#issuecomment-1993148386
